src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -1300,13 +1307,7 @@ async def get_site(self):
 
         return lookup["Client Site Name"]
 
-    @accepts(
-        Dict(
-            'leave_ad',
-            Str('username', required=True),
-            Str('password', required=True, private=True)
-        )
-    )
+    @accepts(Ref('kerberos_username_password'))
```

Comment:
We are missing `returns` decorator here

Reply:
I was planning to handle each plugin separately.